darwin/dataset/remote_dataset_v2.py

```diff
@@ -569,6 +569,80 @@ def _build_image_annotation(
         return build_image_annotation(annotation_file, team_name)
 
     def register(
+        self,
+        object_store: ObjectStore,
+        storage_keys: Union[List[str], Dict[str, List[str]]],
+        fps: Optional[Union[str, float]] = None,
+        multi_planar_view: bool = False,
+        preserve_folders: bool = False,
+        multi_slotted: bool = False,
+    ) -> Dict[str, List[str]]:
+        """
+        Register files from external storage in a Darwin dataset.
+
+        Parameters
+        ----------
+        object_store : ObjectStore
+            Object store to use for the registration.
+        storage_keys : List[str] | Dict[str, List[str]]
+            Either:
+            - Single-slotted items: A list of storage keys
+            - Multi-slotted items: A dictionary with keys as item names and values as lists of storage keys
+        fps : Optional[str], default: None
+            When the uploading file is a video, specify its framerate.
+        multi_planar_view : bool, default: False
+            Uses multiplanar view when uploading files.
+        preserve_folders : bool, default: False
+            Specify whether or not to preserve folder paths when uploading.
+        multi_slotted : bool, default: False
+            Specify whether the items are multi-slotted or not.
+
+        Returns
+        -------
+        Dict[str, List[str]]
+            A dictionary with the list of registered files.
+
+        Raises
+        ------
+        ValueError
+            If the type of ``storage_keys``:
+            - Isn't List[str] when ``multi_slotted`` is False.
+            - Isn't Dict[str, List[str]] when ``multi_slotted`` is True.
+        """
+        if multi_slotted:
+            try:
+                StorageKeyDictModel(storage_keys=storage_keys)  # type: ignore
```

Comment:
Good catch, I've moved the calls to `register_multi_slotted()` and `register_single_slotted()` outside the try/except since we will only throw validation for `StorageKeyDictModel` and `StorageKeyListModel`